### PR TITLE
Implement @:assemblyMeta and @:assemblyStrict

### PR DESCRIPTION
**Pull request by kLabz**
Current limitations:

* Can only be used on the first generated type (enum, class) of a module. Will generate an error for now until this is implemented (might need a more complex implementation).

* Cannot be used on top-level (no package) types. Will generate an error for now, but this could be prevented by adding the attributes to the type itself.

* Does not check for duplicate assembly attributes in a package (pack.A & pack.B defining the same assembly attribute); C# compiler will catch these at compile time.

* Current behavior is being checked at compile time + generation has been checked manually, but proper tests (inspecting the generated assembly or something?) have not been implemented.

* `@:assemblyStrict` will only work with types already included in the compilation, as `@:strict` (see #8395)

See #8347 for original "feature request"

### CODE REVIEW

File: src-json/meta.json [hunk 1]
```diff
@@ -49,6 +49,20 @@
 		"doc": "Allows [] access on an abstract",
 		"targets": ["TAbstract", "TAbstractField"]
 	},
+	{
+		"name": "AssemblyMeta",
+		"metadata": ":assemblyMeta",
+		"doc": "TODO: documentation",
```
Comment (Gama11) [inline src-json/meta.json]: TODO? :)

Comment (kLabz) [reply src-json/meta.json]: Ahah forgot to add this to my TODO list :D

**Comment by Simn** [comment]
Please update the branch and use the `cs` prefix for the metadata. With that I'm fine merging this.

**Comment by Simn** [comment]
Welp, merging your other PR caused a conflict again!

**Comment by kLabz** [comment]
And CI is back to its random fails :D _(but I'll wait for potential real fails)_

**Pull request merged**